### PR TITLE
Feat/balancer v2 lbps abstraction

### PR DESCRIPTION
**Pull request by Stefenon**
Brief comments on the purpose of your changes:

**For Dune Engine V2**
Migrate Balancer's balancer_v2.view_lbps Dune abstraction to a Dune spellbook.
cc: @markusbkoch @mendesfabio @hedgehog-jacek

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
 * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/balancer/ethereum/balancer_v2_ethereum_lbps.sql [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                        '0x6b175474e89094c44da98b954eedeac495271d0f',
+                        '0x88acdd2a6425c3faae4bc9650fd7e27e0bebb7ab',
+                        '0x64aa3364f17a4d01c6f1751fd97c2bd3d7e7f1d5'
+                        ) %} -- WETH, USDC, DAI, MIST & OHM
+
+WITH lbps_call_create AS (
+        {% for create_lbp_contract in create_lbp_contracts %}
+        SELECT * FROM {{ source('balancer_v2_ethereum', create_lbp_contract) }}
+        {% if not loop.last %}
+        UNION ALL
```
Comment (Hosuke) [inline models/balancer/ethereum/balancer_v2_ethereum_lbps.sql:28]: Hi @Stefenon , you may rewrite the jinja loop in the following way:
```suggestion
{% set create_lbp_contracts = [
        source('balancer_v2_ethereum', 'LiquidityBootstrappingPoolFactory_call_create'),
        source('balancer_v2_ethereum', 'NoProtocolFeeLiquidityBootstrappingPoolFactory_call_create')
] %}
{% set lbp_weight_upgrade_contracts = [
        source('balancer_v2_ethereum', 'LiquidityBootstrappingPool_evt_GradualWeightUpdateScheduled'),
        source('balancer_v2_ethereum', 'NoProtocolFeeLiquidityBootstrappingPool_evt_GradualWeightUpdateScheduled')
] %}
{% set non_lbp_tokens = ('0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
                        '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
                        '0x6b175474e89094c44da98b954eedeac495271d0f',
                        '0x88acdd2a6425c3faae4bc9650fd7e27e0bebb7ab',
                        '0x64aa3364f17a4d01c6f1751fd97c2bd3d7e7f1d5'
                        ) %} -- WETH, USDC, DAI, MIST & OHM

WITH lbps_call_create AS (
        {% for create_lbp_contract in create_lbp_contracts %}
        SELECT * FROM {{ create_lbp_contract }}
        {% if not loop.last %}
        UNION ALL
```

File: models/balancer/polygon/balancer_v2_polygon_lbps.sql [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+{% set create_lbp_contracts = ['LiquidityBootstrappingPoolFactory_call_create',
+                                'NoProtocolFeeLiquidityBootstrappingPoolFactory_call_create'] %}
+{% set lbp_weight_upgrade_contracts = ['LiquidityBootstrappingPool_evt_GradualWeightUpdateScheduled',
+                                        'NoProtocolFeeLiquidityBootstrappingPool_evt_GradualWeightUpdateScheduled'] %}
+{% set non_lbp_tokens = ('0x7ceb23fd6bc0add59e62ac25578270cff1b9f619',
+                        '0x2791bca1f2de4661ed88a30c99a7a9449aa84174',
+                        '0x8f3cf7ad23cd3cadbd9735aff958023239c6a063',
+                        ) %} -- WETH, USDC & DAI
+
```
Comment (Hosuke) [inline models/balancer/polygon/balancer_v2_polygon_lbps.sql:21]: Same as above.

**Review by Hosuke — CHANGES_REQUESTED** [review]
Hi @Stefenon, rewriting jinja loops may solve the test failure issues.

**Review by Hosuke — APPROVED** [review]
LGTM.✅

Also we may rewrite `select *` to only select the targeted fields from the sources.

[Sample balancer_v2 ethereum lbps](https://dune.com/queries/2002730?d=1)
[Sample balancer_v2 polygon lbps](https://dune.com/queries/2002733)

File: models/balancer/polygon/balancer_v2_polygon_lbps.sql [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        unique_key = ['pool_id'],
+        post_hook='{{ expose_spells(\'["polygon"]\',
+                                    "project",
+                                    "balancer_v2",
+                                    \'["stefenon"]\') }}'
+    )
+}}
+
+{% set create_lbp_contracts = [
+                                source('balancer_v2_ethereum', 'LiquidityBootstrappingPoolFactory_call_create'),
```
Comment (jeff-dude) [inline models/balancer/polygon/balancer_v2_polygon_lbps.sql:14]: looks like copy/paste error, having wrong blockchain here :)

Comment (jeff-dude) [reply models/balancer/polygon/balancer_v2_polygon_lbps.sql:14]: same in a few places throughout

**Review by jeff-dude — CHANGES_REQUESTED** [review]
minor typo change requested, but great usage of jinja syntax for loops 🤝

File: models/balancer/polygon/balancer_v2_polygon_lbps.sql (hunk 3 above)
Comment (Hosuke) [reply models/balancer/polygon/balancer_v2_polygon_lbps.sql]: just solved the wrong chain issues.

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**